com.ibm.wala.util/src/com/ibm/wala/util/config/ExclusionSpecification.java

```diff
@@ -43,49 +41,49 @@
   }
 
   private Kind kind;
-  private Collection<SetOfClasses> inclusions;
-  private Collection<SetOfClasses> exclusions;
+  private SetOfClasses inclusions;
+  private SetOfClasses exclusions;
   
-  public ExclusionSpecification(Kind kind) {
+  public ExclusionSpecification(Kind kind, SetOfClasses inclusions, SetOfClasses exclusions) {
     this.kind = kind;
-    this.inclusions = new ArrayList<SetOfClasses>();
-    this.exclusions = new ArrayList<SetOfClasses>();
-  }
-  
-  public void addInclusionSet(SetOfClasses set) {
-    if ((kind != Kind.INCL_ONLY) && (kind != Kind.INCL_OVERRIDE_EXCL)){
-      throw new IllegalArgumentException("Illegal attempt to add inclusions when kind is " + kind);
-    }
-    inclusions.add(set);
-  }
-  
-  public void addExclusionSet(SetOfClasses set) {
-    if ((kind != Kind.EXCL_ONLY) && (kind != Kind.INCL_OVERRIDE_EXCL)){
-      throw new IllegalArgumentException("Illegal attempt to add exclusions when kind is " + kind);
+    switch (kind) {
+    case INCL_ONLY :
+      if (inclusions == null || exclusions != null){

```

Comment:
If you think it useful, we could say what actually went wrong, but don't bother if you don't think it helps much.